lib/network_utils.ml

```diff
@@ -896,7 +916,15 @@ module Ovs = struct
 		in
 		let vif_arg =
 			let existing_vifs = List.filter (fun iface -> not (Sysfs.is_physical iface)) (bridge_to_interfaces name) in
-			List.flatten (List.map (fun vif -> ["--"; "--may-exist"; "add-port"; name; vif]) existing_vifs)
+        let ifaces_with_type =
+          let raw = vsctl ~log:false ["--bare"; "-f"; "table"; "--"; "--columns=name,type"; "find"; "interface"; {|type!=\"\"|}] in
+          let lines = String.trim raw |> fun x -> Stringext.split x ~on:'\n' |> List.filter (fun x -> x <> "") in
```

Comment:
This looks like a faithful reimplementation of `Astring.cuts` using stdext's 2.1.0 version.